### PR TITLE
moved table to an include file

### LINKED ISSUE #1093: Improve the .NET Standard table
There are a lot of customer comments about the support table for the .NET Standard versions for this topic:
https://docs.microsoft.com/en-us/dotnet/articles/standard/library#net-platforms-support

We need to improve that on a similar way that was done by @terrajobst on the blog post: 
https://blogs.msdn.microsoft.com/dotnet/2016/09/26/introducing-net-standard/

I'm also thinking that the alias info that is shown in that table would be better presented in this topic instead:
https://docs.microsoft.com/en-us/dotnet/articles/standard/frameworks

### PR DESCRIPTION
**Pull request by mairaw**
Prepping work for issue #1646 to avoid duplicate content to be maintained

@terrajobst does the table look up-to-date to you?

Fixes #1093

### CODE REVIEW

**Comment by terrajobst** [comment]
I sugest to change the row for .NET Framework like this:

```diff
 | Platform Name              | Alias       |        |        |        |        |        |        |        |       |
 |----------------------------|-------------|--------|--------|--------|--------|--------|--------|--------|-------|
 | .NET Standard              | netstandard | 1.0    | 1.1    | 1.2    | 1.3    | 1.4    | 1.5    | 1.6    | 2.0   |
-| .NET Framework             | net         | &rarr; | 4.5    | 4.5.1  | 4.6    | 4.6.1  | 4.6.2  | vNext  | 4.6.1 |
+| .NET Framework             | net         | &rarr; | 4.5    | 4.5.1  | 4.6    | &rarr; | &rarr; | &rarr; | 4.6.1 |
```

Otherwise it looks like 4.6.1 only supports .NET Standard 1.4 and 2.0, but not 1.5 or 1.6. That's no longer true. By claiming support for 2.0, 4.6.1 implicitly supports 1.0 - 2.0, which includes 1.5 and 1.6. The fact that 4.6.2 supported 1.5 and no .NET Framework ever had specific support for 1.6 is irrelevant at this point.

If you do that, you match the table in the .NET Standard repo:

https://github.com/dotnet/standard/blob/master/docs/versions.md

I'd also encourage you add some additional text explaining how to read the table. This is what I wrote in the document linked above:

> The table below lists all versions of .NET Standard:
> 
> * The columns represent .NET Standard versions. The header is a link to a
>   document that shows which APIs got added in that version of .NET Standard.
> * The rows indicate which version of a given .NET platform implements a given
>   .NET Standard version.
> * The arrows indicate that the platform supports a higher version of .NET
>   Standard. For instance, .NET Core 1.0 supports the .NET Standard version 1.6,
>   which is why there are arrows pointing to the right for the lower versions 1.0
>   \- 1.5.
> 
> You can use this table to understand what the highest version of .NET Standard
> is that you can target, based on which .NET platforms you intend to run on. For
> instance, if you want to run on .NET Framework 4.5 and .NET Core 1.0, you can at
> most target .NET Standard 1.1.

**Review by terrajobst — CHANGES_REQUESTED** [review]
I sugest to change the row for .NET Framework like this:

```diff
 | Platform Name              | Alias       |        |        |        |        |        |        |        |       |
 |----------------------------|-------------|--------|--------|--------|--------|--------|--------|--------|-------|
 | .NET Standard              | netstandard | 1.0    | 1.1    | 1.2    | 1.3    | 1.4    | 1.5    | 1.6    | 2.0   |
-| .NET Framework             | net         | &rarr; | 4.5    | 4.5.1  | 4.6    | 4.6.1  | 4.6.2  | vNext  | 4.6.1 |
+| .NET Framework             | net         | &rarr; | 4.5    | 4.5.1  | 4.6    | &rarr; | &rarr; | &rarr; | 4.6.1 |
```

Otherwise it looks like 4.6.1 only supports .NET Standard 1.4 and 2.0, but not 1.5 or 1.6. That's no longer true. By claiming support for 2.0, 4.6.1 implicitly supports 1.0 - 2.0, which includes 1.5 and 1.6. The fact that 4.6.2 supported 1.5 and no .NET Framework ever had specific support for 1.6 is irrelevant at this point.

If you do that, you effectively match the table in the .NET Standard repo:

https://github.com/dotnet/standard/blob/master/docs/versions.md

I'd also encourage you add some additional text explaining how to read the table. This is what I wrote in the document linked above:

> The table below lists all versions of .NET Standard:
> 
> * The columns represent .NET Standard versions. The header is a link to a
>   document that shows which APIs got added in that version of .NET Standard.
> * The rows indicate which version of a given .NET platform implements a given
>   .NET Standard version.
> * The arrows indicate that the platform supports a higher version of .NET
>   Standard. For instance, .NET Core 1.0 supports the .NET Standard version 1.6,
>   which is why there are arrows pointing to the right for the lower versions 1.0
>   \- 1.5.
> 
> You can use this table to understand what the highest version of .NET Standard
> is that you can target, based on which .NET platforms you intend to run on. For
> instance, if you want to run on .NET Framework 4.5 and .NET Core 1.0, you can at
> most target .NET Standard 1.1.

**Comment by mairaw** [comment]
Thanks @terrajobst. I changed the language a bit for the explanation of the table, especially the part about the arrows which is what causes the most confusion. Even if we say that .NET Core 1.0 supports .NET Standard 1.6, it's not clear you can target the lower versions that have the arrows. What do you think?

Also, I wanted to add the part about [choosing the version to target](https://github.com/dotnet/standard/blob/master/docs/versions.md#how-do-i-know-which-net-standard-version-i-should-target), but I had a question about the part that says that generally speaking you'd want to target the lowest version. So after you figure out the highest you can target, then you need to analyze the APIs you want to use and pick the lowest?

Thanks!

File: docs/standard/library.md
Comment (terrajobst) [inline docs/standard/library.md]: While I like the way you phrased it, I wonder whether *compatibility matrix* makes it sound too complicated. I'm not a doc writer, but every time I hear *matrix* it reminds me of math, exams, and frustration :-)

File: docs/includes/net-standard-table.md
Comment (terrajobst) [inline docs/includes/net-standard-table.md]: Should the .NET Standard row effectively become the header? This would underline stronger that this table really shows the relationship between all platforms and .NET Standard. Otherwise one can read the table in such a way that all platforms are somehow related to each other -- which isn't the case. In other words, .NET Standard isn't a peer of the other rows; it is the driver for the column.

File: docs/standard/library.md
Comment (terrajobst) [inline docs/standard/library.md]: > even though you can target a lower version of .NET Standard for that given platform.

❤️

**Comment by terrajobst** [comment]
> [@mairaw] What do you think?

Looks great! Do you think it would be useful to incorporate these two section as well? The might be useful for understanding how .NET Standard is supposed to work and what you can expect.

> ## How do I know which .NET Standard version I should target?
> 
> When choosing a .NET Standard version, you should consider this trade-off:
> 
> * The higher the version, the more APIs are available to you.
> * The lower the version, the more platforms implement it.
> 
> So generally speaking, you should target the lowest version you get away with.
> To inform your decision, you should consider the compatibility matrix and the
> APIs additions linked from the table above.
> 
> ## Versioning rules
> 
> There are two primary versioning rules:
> 
> 1. **Additive**. .NET Standard versions are logically concentric circles: higher
>    versions incorporate all APIs from previous versions. There are no breaking
>    changes between versions.
> 2. **Immutable**. Once shipped, .NET Standard versions are frozen.
> 
> New APIs will first become available in specific .NET platforms, such as .NET
> Core. If the [.NET Standard review board][netstandard-board] believes the new
> APIs should be made available everywhere, they'll be added in a new .NET
> Standard version.

> [@mairaw] So after you figure out the highest you can target, then you need to analyze the APIs you want to use and pick the lowest?

Yes, and I follow a strict scientific process:

1. Target next lower version.
2. Does it still build?
    - Yes? Go to 1.
    - No? Retarget to the higher version, stop.

Once we have more versions, you can modify this process to use a binary search instead ;-)

In all seriousness, it would be great if we had tooling that does that automatically.

**Review by terrajobst — APPROVED** [review]

File: docs/standard/library.md
Comment (mairaw) [inline docs/standard/library.md]: I think that's the bit that people get lost the most with the arrows 😄

File: docs/standard/library.md
Comment (mairaw) [inline docs/standard/library.md]: I certainly don't wanna cause that. I'll rewrite it.

**Comment by mairaw** [comment]
I'll add that. I thought there were some useful info there but wasn't sure what the process was to pick up the .NET Standard version so wanted to confirm that before adding. Thanks @terrajobst!

File: docs/includes/net-standard-table.md
Comment (weshaggard) [inline docs/includes/net-standard-table.md]: I think we should add some extra instructions on how to read the table here. See https://github.com/dotnet/standard/pull/243#issuecomment-286926245 for some suggestions from @sam-wheat which I think are pretty good.

File: docs/includes/net-standard-table.md
Comment (weshaggard) [inline docs/includes/net-standard-table.md]: After many points of confusion for folks I think we should do away with the arrows and just explicitly list the version in each slot.

Comment (mairaw) [inline docs/includes/net-standard-table.md]: @weshaggard the information on how to read the table is in the .NET Standard topic itself. Since this information was reused in other topics, I moved to this "include" file so it can be reused in several topics if needed. But I'll change the instructions a bit. New commit coming soon.

**Comment by mairaw** [comment]
@weshaggard @terrajobst the topic has been updated. Please take a look. I've also changed the terminology from .NET Standard Library to just .NET Standard. Since this topic has been written a long time ago, if you find any inaccuracies, please let me know. I'd like at some point to simplify this topic a bit more, but I won't have time to do this with this PR.

File: docs/includes/net-standard-table.md
Comment (weshaggard) [inline docs/includes/net-standard-table.md]: I would suggest putting the instructions for how to read the table in the common table file as well so they go along with the table wherever we import it so folks have a common way to read it.

File: docs/standard/library.md
Comment (weshaggard) [inline docs/standard/library.md]: "The column header indicates the .NET Standard version that supports your target"  - .NET Standards don't support a target a target platform supports a .NET Standard. So I think I would change this to "Find the column header aligned with the platform version you chose and that column header will indicate the highest .NET Standard version that platform supports.

File: docs/standard/library.md
Comment (weshaggard) [inline docs/standard/library.md]: Should we prefix "platform" with ".NET Platform" as platform can sometimes be ambiguous.

File: docs/standard/library.md
Comment (weshaggard) [inline docs/standard/library.md]: I think I would put all the information from the table to this point in the shared table file so it gets imported with the table.

File: docs/standard/library.md
Comment (weshaggard) [inline docs/standard/library.md]: While I don't mind I'm curious why the change from ".NET Standard Library" to ".NET Standard" throughout the document? Also the document name itself is called "library.md" :)

File: docs/includes/net-standard-table.md
Comment (weshaggard) [inline docs/includes/net-standard-table.md]: Did you see how I marked out the versions in the .NET Framework row in my PR? that is because the tools are changing and what you have here represents tools that have not released yet. I would suggest we mark out the versions here and add a note about the tools change so folks at least have the information.

**Comment by sam-wheat** [comment]
>Find the column in that row that indicates your version starting from right to left. 

what is the reason for reading from right to left?

> you should pick the common version among them, which would be the smaller version. 

If one is smaller they can't be common, right?  Shouldn't it be something like "Repeat the process for each platform you want to target and take the lowest standard".

**Comment by sam-wheat** [comment]
Also, comparing versions of different platforms is not meaningful.. so I don't understand that sentence at all.

File: docs/standard/library.md
Comment (mairaw) [inline docs/standard/library.md]: I thought that .NET Standard Library was the old terminology for this and we got some feedback from customers asking about it too. I'm happy to revert this if needed. This topic was last updated last June.

Yes, I know about the file name, I was considering if I should change it or not. :-(

File: docs/standard/library.md
Comment (mairaw) [inline docs/standard/library.md]: Changed the sentence to say "the .NET Standard version that your target supports" instead. I hope this fixes the inaccuracy.

File: docs/standard/library.md
Comment (mairaw) [inline docs/standard/library.md]: Done

File: docs/includes/net-standard-table.md
Comment (mairaw) [inline docs/includes/net-standard-table.md]: Ok now that I read the part of the .NET 2.0 tooling. So I'm thinking we should keep the current state of the tooling in the docs and just change it in the standard repo for now with a note that this is changing. Thoughts?

**Comment by mairaw** [comment]
@sam-wheat if you're trying to find the highest version, I think you should do right from left. Let's say you want to target .NET Core 1.0, first one on the right is 2.0, then 1.0 comes with 1.6. Makes sense?

I'm changing the bit about the common version.

File: docs/standard/library.md
Comment (mairaw) [inline docs/standard/library.md]: @weshaggard I put everything that was part of how to read the table. The information on how to find the highest I left it out, since the other topic that uses this has its own info about what to target, etc.

**Comment by sam-wheat** [comment]
@mairaw In my opinion its counter-intuitive.  More correct however is that its unnecessary since it does not make a difference one way or the other.  Unnecessary instructions lead the reader to wonder what he/she is missing and are source of confusion.

File: docs/standard/library.md
Comment (sam-wheat) [inline docs/standard/library.md]: > The version number in each cell indicates the *minimum* version of the platform you'll need to implement that .NET Standard version. 

One more comment if you don't mind:

I think ".NET Standard version" should be ".NET Standard" since each standard is unique and has no versions.

Other than that, that instruction is perfect - short, simple, non-ambiguous, all inclusive 🥇

File: docs/standard/library.md
Comment (Nirmal4G) [inline docs/standard/library.md]: @mairaw can you also update this table as in my PR #1755 and this #243

**Review by Nirmal4G — CHANGES_REQUESTED** [review]
Please check these suggestions...

File: docs/includes/net-standard-table.md
Comment (Nirmal4G) [inline docs/includes/net-standard-table.md]: In the net-standard-table use one row for each tooling version for .net framework.

---

| .NET Standard                                            | 1.0   | 1.1    | 1.2   | 1.3   | 1.4    | 1.5     | 1.6       | 2.0     |
|----------------------------------------------|------|-------|------|------|-------|-------|---------|-------|
| .NET Framework (with tooling 1.0)            | 4.5   | 4.5    | 4.5.1 | 4.6   | 4.6.1 | 4.6.2  | vNext   | 4.6.1 |
| .NET Framework (with tooling 2.0)            | 4.5   | 4.5    | 4.5.1 | 4.6   | 4.6.1 | 4.6.1   | 4.6.1    | 4.6.1 |

---

This makes it much more clearer than before.
It makes clear for me at least :-)

Any thoughts?

File: docs/standard/library.md
Comment (Nirmal4G) [inline docs/standard/library.md]: Can also add `.NET Standard` as a keyword too!

File: docs/standard/library.md
Comment (weshaggard) [inline docs/standard/library.md]: I'm indifferent about this I will defer to the PM's @terrajobst @richlander about the terminology.

File: docs/includes/net-standard-table.md
Comment (weshaggard) [inline docs/includes/net-standard-table.md]: I personally think we can make this public document have both the current and the future expected information. It will give folks a heads up that there are some changes coming and something to think about, so I would put it both here and in the standard repo.

File: docs/includes/net-standard-table.md
Comment (weshaggard) [inline docs/includes/net-standard-table.md]: I'm not sure it is worth 2 rows I think just crossing out and adding a note in the table is good enough but in either case I think the information should be included.

File: docs/standard/library.md
Comment (sam-wheat) [inline docs/standard/library.md]: I will add a comment on this since this was a big source of confusion for me.  
A "library" is code you execute.
A "standard" is something that describes what you can expect the library to do.
The term "standard library" is a misnomer in this context since it mixes those two concepts.
For this converstation the libraries are .Net core, .Net Framework, etc.  There is no "standard library".
I can understand the term "Standard Libraries" which may mean all the libraries that implment the standard but I think care should be taken not to imply there is a library different from the others in that it is itself the standard.

Comment (terrajobst) [inline docs/standard/library.md]: I like the rename from ".NET Standard Library" to just ".NET Standard".

We should align the name with how we present it in the UX. We don't refer to ".NET Standard Library" anywhere in our UX anymore (yes, the package is still still `NetStandard.Library` but we basically go out of our way to hide that. Also, I've seen people getting confused about it because they don't know whether ".NET Standard Library" refers to their code or .NET Standard itself. Furthermore, the name should align with how we present it conceptually to our developers and we present it as a standard .NET platforms have to implement.

File: docs/standard/library.md
Comment (terrajobst) [inline docs/standard/library.md]: I suggest to leave the note out. The correct TFM representation of the profile isn't that important and will confuse it even more, IMHO.

File: docs/standard/library.md
Comment (svick) [inline docs/standard/library.md]: @sam-wheat I don't think it's a misnomer. In other languages, "standard library" describes exactly this: an interface, potentially with multiple implementations.

The most well known is probably [the C standard library](https://en.wikipedia.org/wiki/C_standard_library), with implementations like glibc, MSVCRT.DLL or musl.

Comment (sam-wheat) [inline docs/standard/library.md]: @svick you are correct which is why the issue the issue becomes confusing.  In the case of c a library became a standard because it was the most common one.  I mentioned the term was a misnomer in this context because .net now has several libraries none of which  is any more or less standard than the other.  In this context I think it is helpful to think of the standard as a common attribute which is a different entity from the libraries that implement it (an interface for those who think of it technically).

Comment (sam-wheat) [inline docs/standard/library.md]: @svick you are correct which is why the issue the issue becomes confusing.  In the case of c a library became a standard because it was the most common one.  I mentioned the term was a misnomer in this context because .net now has several libraries none of which  is any more or less standard than the other.  I think it is helpful to think of the standard as a common attribute which is a different entity from the libraries that implement it (an interface for those who think of it technically).   In this context when we speak of The Standard it should be clear we are NOT referring to an implementation.

File: docs/includes/net-standard-table.md
Comment (Nirmal4G) [inline docs/includes/net-standard-table.md]: Seconded! but the whole purpose of docs is to be **not confusing** and this is not confusing. At least for the time being when people migrates to the new tooling and are aware of the changes.

File: docs/standard/library.md
Comment (Nirmal4G) [inline docs/standard/library.md]: Then how about we put them in the cells like _Profile7_...?

File: docs/standard/library.md
Comment (Nirmal4G) [inline docs/standard/library.md]: I'd like to remove that awful `Library` part as @terrajobst mentioned it's just a Standard and with @sam-wheat 's explanation, the name `.NET Standard or NETStandard` is most apt in this conversation!

File: docs/toc.md
Comment (Nirmal4G) [inline docs/toc.md]: And also the file name can be `standard/`
1. `info.md`
2. `index.md`
3. `net-standard.md` (confusion with path name)

for me I'll go with `2`

**Comment by karelz** [comment]
@mairaw just curious what is blocking this PR ... just got some feedback on gitter about https://github.com/dotnet/standard/pull/243 being better ... can we match it in docs?

**Comment by BillWagner** [comment]
@karelz I fixed the merge conflicts.  When this finishes building, I'll merge it.

**Review by Nirmal4G — COMMENTED** [review]
Any updates on proposed changes?

**Comment by BillWagner** [comment]
@Nirmal4G When I looked at all the comments and changes, I thought everything had been addressed. Do you see anything I missed?

File: docs/standard/library.md
Comment (BillWagner) [inline docs/standard/library.md]: updated.

File: docs/includes/net-standard-table.md
Comment (BillWagner) [inline docs/includes/net-standard-table.md]: updated.

Comment (Nirmal4G) [inline docs/includes/net-standard-table.md]: @BillWagner what about this, are we added it?

@weshaggard said It's not necessary, but when I showed it to newbies, the above table is more clear to them than the one already there, with *'s confusing them.

I think It's better to tell the tooling explicitly!

**Comment by Nirmal4G** [comment]
@BillWagner Thanks

That's all I have, I'll see if something else needs me when I go through the docs!

**Review by Nirmal4G — APPROVED** [review]

**Review by Nirmal4G — APPROVED** [review]

File: docs/includes/net-standard-table.md
Comment (BillWagner) [inline docs/includes/net-standard-table.md]: I did add it.

**Pull request merged**
